libs/groovylib/src/mill/groovylib/JavaMavenModuleWithGroovyTests.scala

```diff
@@ -0,0 +1,16 @@
+package mill.groovylib
+
+import mill.*
+import mill.javalib.{JavaModule, MavenModule}
+
+/**
+ * Convenience trait for projects using Java for production and Groovy for tests in a Maven setup
+ */
```

Comment:
You should add more motivation, why this might be preferable over just adding `GroovyModule` (e.g. to avoid implicit groovy deps in the module).